File: android/app/src/main/java/com/pair/darb/agent/data/local/prefs/PrefsManager.java
package com.pair.darb.agent.data.local.prefs;

import android.content.Context;
import android.content.SharedPreferences;

public class PrefsManager {

    private static final String PREF_FILE = "darb_agent_prefs";

    private static final String KEY_DEVICE_ID       = "device_id";
    private static final String KEY_DRIVER_ID       = "driver_id";
    private static final String KEY_DRIVER_NAME     = "driver_name";
    private static final String KEY_PLATFORM        = "platform";
    private static final String KEY_TENANT_ID       = "tenant_id";
    private static final String KEY_API_TOKEN       = "api_token";
    private static final String KEY_IS_ENROLLED     = "is_enrolled";
    private static final String KEY_IS_ON_SHIFT     = "is_on_shift";
    private static final String KEY_CURRENT_SHIFT_ID = "current_shift_id";

    private static volatile PrefsManager instance;

    private final SharedPreferences prefs;

    private PrefsManager(Context context) {
        prefs = context.getApplicationContext()
                .getSharedPreferences(PREF_FILE, Context.MODE_PRIVATE);
    }

    public static PrefsManager getInstance(Context context) {
        if (instance == null) {
            synchronized (PrefsManager.class) {
                if (instance == null) {
                    instance = new PrefsManager(context);
                }
            }
        }
        return instance;
    }

    // ── deviceId ──────────────────────────────────────────────────────────────

    public String getDeviceId() {
        return prefs.getString(KEY_DEVICE_ID, null);
    }

    public void setDeviceId(String deviceId) {
        prefs.edit().putString(KEY_DEVICE_ID, deviceId).apply();
    }

    // ── driverId ──────────────────────────────────────────────────────────────

    public String getDriverId() {
        return prefs.getString(KEY_DRIVER_ID, null);
    }

    public void setDriverId(String driverId) {
        prefs.edit().putString(KEY_DRIVER_ID, driverId).apply();
    }

    // ── driverName ────────────────────────────────────────────────────────────

    public String getDriverName() {
        return prefs.getString(KEY_DRIVER_NAME, null);
    }

    public void setDriverName(String driverName) {
        prefs.edit().putString(KEY_DRIVER_NAME, driverName).apply();
    }

    // ── platform ──────────────────────────────────────────────────────────────

    public String getPlatform() {
        return prefs.getString(KEY_PLATFORM, null);
    }

    public void setPlatform(String platform) {
        prefs.edit().putString(KEY_PLATFORM, platform).apply();
    }

    // ── tenantId ──────────────────────────────────────────────────────────────

    public String getTenantId() {
        return prefs.getString(KEY_TENANT_ID, null);
    }

    public void setTenantId(String tenantId) {
        prefs.edit().putString(KEY_TENANT_ID, tenantId).apply();
    }

    // ── apiToken ──────────────────────────────────────────────────────────────

    public String getApiToken() {
        return prefs.getString(KEY_API_TOKEN, null);
    }

    public void setApiToken(String apiToken) {
        prefs.edit().putString(KEY_API_TOKEN, apiToken).apply();
    }

    // ── isEnrolled ────────────────────────────────────────────────────────────

    public boolean isEnrolled() {
        return prefs.getBoolean(KEY_IS_ENROLLED, false);
    }

    public void setEnrolled(boolean enrolled) {
        prefs.edit().putBoolean(KEY_IS_ENROLLED, enrolled).apply();
    }

    // ── isOnShift ─────────────────────────────────────────────────────────────

    public boolean isOnShift() {
        return prefs.getBoolean(KEY_IS_ON_SHIFT, false);
    }

    public void setOnShift(boolean onShift) {
        prefs.edit().putBoolean(KEY_IS_ON_SHIFT, onShift).apply();
    }

    // ── currentShiftId ────────────────────────────────────────────────────────

    public String getCurrentShiftId() {
        return prefs.getString(KEY_CURRENT_SHIFT_ID, null);
    }

    public void setCurrentShiftId(String shiftId) {
        prefs.edit().putString(KEY_CURRENT_SHIFT_ID, shiftId).apply();
    }

    // ── helpers ───────────────────────────────────────────────────────────────

    /** Clear all stored preferences (e.g. on logout / un-enroll). */
    public void clearAll() {
        prefs.edit().clear().apply();
    }
}

File: android/app/src/main/java/com/fleetpulse/agent/data/local/prefs/PrefsManager.kt
package com.fleetpulse.agent.data.local.prefs

import android.content.SharedPreferences
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class PrefsManager @Inject constructor(
    private val prefs: SharedPreferences
) {
    var deviceToken: String?
        get() = prefs.getString(KEY_DEVICE_TOKEN, null)
        set(value) = prefs.edit().putString(KEY_DEVICE_TOKEN, value).apply()

    var baseUrl: String
        get() = prefs.getString(KEY_BASE_URL, DEFAULT_BASE_URL) ?: DEFAULT_BASE_URL
        set(value) = prefs.edit().putString(KEY_BASE_URL, value).apply()

    var deviceId: String?
        get() = prefs.getString(KEY_DEVICE_ID, null)
        set(value) = prefs.edit().putString(KEY_DEVICE_ID, value).apply()

    var tenantId: String?
        get() = prefs.getString(KEY_TENANT_ID, null)
        set(value) = prefs.edit().putString(KEY_TENANT_ID, value).apply()

    var driverId: String?
        get() = prefs.getString(KEY_DRIVER_ID, null)
        set(value) = prefs.edit().putString(KEY_DRIVER_ID, value).apply()

    var driverName: String?
        get() = prefs.getString(KEY_DRIVER_NAME, null)
        set(value) = prefs.edit().putString(KEY_DRIVER_NAME, value).apply()

    var isRegistered: Boolean
        get() = prefs.getBoolean(KEY_IS_REGISTERED, false)
        set(value) = prefs.edit().putBoolean(KEY_IS_REGISTERED, value).apply()

    var isClockedIn: Boolean
        get() = prefs.getBoolean(KEY_IS_CLOCKED_IN, false)
        set(value) = prefs.edit().putBoolean(KEY_IS_CLOCKED_IN, value).apply()

    var currentShiftId: String?
        get() = prefs.getString(KEY_CURRENT_SHIFT_ID, null)
        set(value) = prefs.edit().putString(KEY_CURRENT_SHIFT_ID, value).apply()

    var vehicleId: String?
        get() = prefs.getString(KEY_VEHICLE_ID, null)
        set(value) = prefs.edit().putString(KEY_VEHICLE_ID, value).apply()

    var clockInTime: String?
        get() = prefs.getString(KEY_CLOCK_IN_TIME, null)
        set(value) = prefs.edit().putString(KEY_CLOCK_IN_TIME, value).apply()

    fun clear() {
        prefs.edit().clear().apply()
    }

    companion object {
        const val PREFS_NAME = "fleetpulse_prefs"
        const val DEFAULT_BASE_URL = "http://10.0.2.2:8000"

        private const val KEY_DEVICE_TOKEN = "device_token"
        private const val KEY_BASE_URL = "base_url"
        private const val KEY_DEVICE_ID = "device_id"
        private const val KEY_TENANT_ID = "tenant_id"
        private const val KEY_DRIVER_ID = "driver_id"
        private const val KEY_DRIVER_NAME = "driver_name"
        private const val KEY_IS_REGISTERED = "is_registered"
        private const val KEY_IS_CLOCKED_IN = "is_clocked_in"
        private const val KEY_CURRENT_SHIFT_ID = "current_shift_id"
        private const val KEY_VEHICLE_ID = "vehicle_id"
        private const val KEY_CLOCK_IN_TIME = "clock_in_time"
    }
}
